src/app/haku/[oid]/hakukohde/[hakukohde]/sijoittelun-tulokset/components/other-actions-hakukohde-button.tsx

```diff
@@ -0,0 +1,247 @@
+  osoitetarraDocumentId: string | null;
+  tulosDocumentId: string | null;
+  sijoitteluajoId: string;
+}) => {
+  const { t } = useTranslations();
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [hyvaksymiskirje, setHyvaksymiskirjeDocument] = useState<string | null>(
+    hyvaksymiskirjeDocumentId,
+  );
+  const [osoitetarraDocument, setOsoitetarraDocument] = useState<string | null>(
+    osoitetarraDocumentId,
+  );
+
+  const open = Boolean(anchorEl);
+  const buttonId = `other-actions-hakukohde-menu`;
+  const { addToast } = useToaster();
+
+  const showMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const closeMenu = () => setAnchorEl(null);
+
+  const openAcceptedLetterTemplateModal = async () => {
+    showModal(AcceptedLetterTemplateModal, {
+      title: 'kirje-modaali.otsikko-hyvaksymiskirjeet',
+      hakukohde: hakukohde,
+      template: 'hyvaksymiskirje',
+      sijoitteluajoId,
+      setDocument: setHyvaksymiskirjeDocument,
+    });
+    closeMenu();
+  };
+
+  const openNonAcceptedLetterTemplateModal = async () => {
+    showModal(NonAcceptedLetterTemplateModal, {
+      title: 'kirje-modaali.otsikko-ei-hyvaksymiskirjeet',
+      hakukohde: hakukohde,
+      template: 'jalkiohjauskirje',
+      sijoitteluajoId,
+    });
+    closeMenu();
+  };
+
+  const openOsoitetarratModal = async () => {
+    showModal(ProgressModal, {
+      title: 'Osoitetarrojen muodostaminen',
+      defaultFileName: 'osoitetarrat.pdf',
+      progressMessage: 'Osoitetarroja muodostetaan....',
+      setDocument: setOsoitetarraDocument,
+      functionToMutate: () =>
+        luoOsoitetarratHakukohteessaHyvaksytyille({
+          sijoitteluajoId,
+          hakukohde,
+        }),
+    });
+    closeMenu();
+  };
+
+  const sendVastaanottoposti = async () => {
+    try {
+      const data = await sendVastaanottopostiHakukohteelle(hakukohde.oid);
+      if (!data || data.length < 1) {
+        addToast({
+          key: 'vastaanottoposti-hakemus-empty',
+          message:
+            'sijoittelun-tulokset.toiminnot.vastaanottoposti-hakukohteelle-ei-lahetettavia',
+          type: 'error',
+        });
+      } else {
+        addToast({
+          key: 'vastaanottoposti-hakemus',
+          message:
+            'sijoittelun-tulokset.toiminnot.vastaanottoposti-hakukohteelle-lahetetty',
+          type: 'success',
+        });
+      }
+    } catch (e) {
+      addToast({
+        key: 'vastaanottoposti-hakemus-virhe',
+        message:
+          'sijoittelun-tulokset.toiminnot.vastaanottoposti-hakukohteelle-virhe',
+        type: 'error',
+      });
+      console.error(e);
+    }
+    closeMenu();
+  };
+
+  const openDocument = async (documentId: string | null) => {
+    if (documentId) {
+      window.open(
+        configuration.lataaDokumenttiUrl({ dokumenttiId: documentId }),
+      );
+    }
+    closeMenu();
+  };
```

Comment:
Tätä komponenttia selkeyttäisi aika paljon, jos eri MenuItem-palasia jakaisi omiksi komponenteikseen, jolloin esim. näitä handlereita saisi lähemmäksi komponentin koodia, jossa niitä oikeasti käytetään.